### PR TITLE
As16 - scoring functions

### PR DESCRIPTION
**Pull request by glennclarence**
integration of the transform, reduction and Nl functions.
started to write more explanatory comments for each commits (started with the last one ;))
By pointing out the major changes it should be more efficient for you to folloe my changes.

also tried to introduce more structure to the commits so that each commit is one categorised step

### CODE REVIEW

File: src/model/neighborlist_modes.h [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+					fRec.x += fEl.x;;
+					fRec.y += fEl.y;
+					fRec.z += fEl.z;
+					eRec  -=  eEl;
+
+
+				}
+
+				outRec_fx[nIdx]+=fRec.x;
+				outRec_fx[nIdx]+=fRec.y;
```
Comment (UmbrellaSampler) [inline src/model/neighborlist_modes.h]: must be outRec_fy, outRec_fz. not outRec_fx in all cases

File: src/model/neighborlist_modes.h [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+		REAL const* RecPosX,
+		REAL const* RecPosY,
+		REAL const* RecPosZ,
+		REAL const* LigPosX,
+		REAL const* LigPosY,
+		REAL const* LigPosZ,
+		REAL* outLig_fx,
+		REAL* outLig_fy,
+		REAL* outLig_fz,
+		REAL* outLig_E,
```
Comment (UmbrellaSampler) [inline src/model/neighborlist_modes.h:34]: Are you sure that outLig_E  and outRec_E must be differentiated? Is there not only one Energy in the system?

File: src/model/neighborlist_modes.h [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+					fRec.x += fEl.x;;
+					fRec.y += fEl.y;
+					fRec.z += fEl.z;
+					eRec  -=  eEl;
+
+
+				}
+
+				outRec_fx[nIdx]+=fRec.x;
```
Comment (UmbrellaSampler) [inline src/model/neighborlist_modes.h]: code style: please put spaces before and after the operator: outRec_fx[nIdx] += fRec.x;

File: src/model/reduction_modes.h [hunk 4]
```diff
@@ -53,5 +53,57 @@ void reduceModeForce(
...
+		unsigned const& numModes,
+		REAL* result
+		)
+{
+	//TODO: think about passing protein to function with member "isreceptor"to determine rotation
+	//rotate forces into ligand frame
+	for( int i=0; i<numModes;i++){result[i]=0;}
+	for (unsigned i = 0; i < numAtoms; ++i) {
+		for(int mode=0;mode<numModes;mode++){
+				result[mode] -= forceX[i]*modeX[i*numModes+mode]+forceY[i]*modeY[i*numModes+mode]+forceZ[i]*modeZ[i*numModes+mode];
```
Comment (UmbrellaSampler) [inline src/model/reduction_modes.h]: please wrap the line. I can't see all of it here in github :/

File: src/model/reduction_modes.h [hunk 5]
```diff
@@ -53,5 +53,57 @@ void reduceModeForce(
...
+ * The forceconstant correspoonds to the magnitude of the modevector for each mode.
+ */
+template<typename REAL>
+void correctModeForce(
+		const REAL* modeForceConstant,
+		unsigned const& numModes,
+		REAL* delta
+		)
+{
+	const REAL factor=4.0;
```
Comment (UmbrellaSampler) [inline src/model/reduction_modes.h]: int c++11 make it a constexpr.: constexpr REAL factor=4.0; We later can put these constants into a configuration header

please insert spaces before and after operators in general (here: "y = x"  )

File: src/model/transform_modes.h [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+
+#include "transform.h"
+
+ namespace as{
+
+
+
+
+ /**
+  * @brief: to change from the receptor system to the ligand system the translational and rotational DOF have to be "inverted".
```
Comment (UmbrellaSampler) [inline src/model/transform_modes.h]: line wrap...

File: src/model/transform_modes.h [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+	invertedDOF._6D.pos=ligandDOF._6D.pos.inv();
+	const RotMat<REAL> rotMat=euler2rotmat(ligandDOF._6D.ang.x, ligandDOF._6D.ang.y, ligandDOF._6D.ang.z).getInv();
+	invertedDOF._6D.pos=rotMat*invertedDOF.pos;
+return invertedDOF;
+}
+
+
+
+/**
+ * @brief: In addition to the normal transform function rotate_translate_deform also applies mode deformation to the coodinates.
```
Comment (UmbrellaSampler) [inline src/model/transform_modes.h]: line wrap

**Review by UmbrellaSampler — CHANGES_REQUESTED** [review]
why better than the former pull requests! Good job!

File: src/model/neighborlist_modes.h (hunk 2 above)
Comment (glennclarence) [reply src/model/neighborlist_modes.h:34]: while I was writing it i realized that that's most likely the case. Id agree

**Review by UmbrellaSampler — APPROVED** [review]
Alright :)

**Comment by UmbrellaSampler** [comment]
[AS 16 - implementierunge der scoring funktionen für modes (transform+deform, NL reduction inkl. receptor, DOFinverter, reduce modes, correctModeforces)](https://trello.com/c/LM0MzMKn/27-as-16-implementierunge-der-scoring-funktionen-f%C3%BCr-modes-transformdeform-nl-reduction-inkl-receptor-dofinverter-reduce-modes-corr)

**Pull request merged**